docs/sdk/examples/assets/pastefromoffice/pastefromoffice.js

```diff
@@ -0,0 +1,164 @@
+var sources = [
+    {
+        name: 'Google Docs',
+        link: 'pastefromgoogledocs.html',
+        linkName: 'Paste from Google Docs',
+        regex: /id=(\"|\')?docs\-internal\-guid\-/,
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Word',
+        link: 'pastefromword.html',
+        linkName: 'Paste from Word',
+        regex: getProperWordRegex(),
+        isAlreadyShow: false
+    },
+    {
+        name: 'Microsoft Excel',
+        link: 'pastefromexcel.html',
+        linkName: 'Paste from Excel',
+        regex: /<meta\s*name=generator\s*content="Microsoft\s*Excel\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    },
+    {
+        name: 'LibreOffice',
+        link: 'pastefromlibreoffice.html',
+        linkName: 'Paste from LibreOffice',
+        regex: /<meta\s+name=["']?generator["']?\s+content="LibreOffice\s*\d+"?\/?/gmi,
+        isAlreadyShow: false
+    }
+];
+var instances = getInstances();
+// A state variable that indicates if the clipboard notification has been seen.
+// We use the variable for displaying the notification only once per demo.
+let hasNotificationBeenSeen = false;
+
+instances.forEach( function ( instanceName ) {
+    var editor = CKEDITOR.instances[ instanceName ];
+
+    if ( !editor ){
+        return;
+    }
+
+    editor.on( 'instanceReady', function() {
+        editor.on( 'paste', function( evt ) {
+            var pasteData = CKEDITOR.plugins.pastetools.getClipboardData( evt.data, 'text/html' );
+
+            if( !pasteData ){
+                return;
+            }
+
+            var sourceIndex = getPastedSourceIndex( pasteData );
+            // Show notification only once per pasted source format ( Word, Excel ).
+            if( sourceIndex !== -1 && !hasNotificationBeenSeen && !sources[ sourceIndex ].isAlreadyShow ) {
+                var notice = document.querySelector('.main__notification.notice');
+
+                // Remove notification when user trying to paste content from other source than previous. e.g. Word and Excel.
+                if( notice ){
+                    notice.remove();
+                }
+
+                createClipboardInputNotification( sourceIndex );
+                sources[ sourceIndex ].isAlreadyShow = true;
+            }
```

Comment:
I have a feeling that the logic responsible for getting a specific handler is a bit complicated. I suppose we could just use for loop for that?

```js
for ( var i = 0; i < sources.length; i++ ) {
  var source = sources[ i ];
  if ( source.regex.test( htmlString ) ) {
    createClipbardInputNotification( source );
    break;
  }
}
```
so we don't have to rely on some magic indexes of `sources` array.